### ISSUE TITLE
golang grabs a random tag if the repo has been forked

### ISSUE THREAD

**Issue by KyleSanderson** [dependabot/dependabot-core#7742]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

gomod

### Language version

1.20

### Manifest location and content before the Dependabot update

https://github.com/autobrr/autobrr/pull/1036

### dependabot.yml content

https://github.com/autobrr/autobrr/pull/1028

### Updated dependency

https://github.com/autobrr/go-rtorrent

### What you expected to see, versus what you actually saw

It's using some weird old tag that doesn't exist anywhere, and it's clearly failing the actions and doing nothing.

### Native package manager behavior

doesn't happen.

**Comment by jakecoffman** [comment]
Dependabot uses this command to find versions: 

```console
$ go list -m -versions -json github.com/autobrr/go-rtorrent
```

In the logs we can see the resulting calls:

```
updater | 2023/08/03 22:06:36 INFO <job_704829211> Checking if github.com/autobrr/go-rtorrent 1.0.1 needs updating
  proxy | 2023/08/03 22:06:36 [592] GET https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/list
  proxy | 2023/08/03 22:06:37 [592] 200 https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/list
  proxy | 2023/08/03 22:06:37 [594] GET https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/v1.9.0.info
  proxy | 2023/08/03 22:06:37 [594] 200 https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/v1.9.0.info
  proxy | 2023/08/03 22:06:37 [596] GET https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/v1.9.0.mod
  proxy | 2023/08/03 22:06:37 [596] 200 https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/v1.9.0.mod
  proxy | 2023/08/03 22:06:37 [598] GET https://proxy.golang.org:443/sumdb/sum.golang.org/supported
  proxy | 2023/08/03 22:06:37 [598] 404 https://proxy.golang.org:443/sumdb/sum.golang.org/supported
  proxy | 2023/08/03 22:06:37 [600] GET https://sum.golang.org:443/lookup/github.com/autobrr/go-rtorrent@v1.9.0
  proxy | 2023/08/03 22:06:37 [600] 200 https://sum.golang.org:443/lookup/github.com/autobrr/go-rtorrent@v1.9.0
  proxy | 2023/08/03 22:06:37 [602] GET https://sum.golang.org:443/tile/8/0/x067/914
  proxy | 2023/08/03 22:06:37 [602] 200 https://sum.golang.org:443/tile/8/0/x067/914
updater | 2023/08/03 22:06:37 INFO <job_704829211> Latest version is 1.9.0
```

So it appears at the time the latest version was 1.9.0, even though I can see 1.10.0 there now: https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/list

**Comment by KyleSanderson** [comment]
> Dependabot uses this command to find versions:
> 
> ```
> $ go list -m -versions -json github.com/autobrr/go-rtorrent
> ```
> 
> In the logs we can see the resulting calls:
> 
> ```
> updater | 2023/08/03 22:06:36 INFO <job_704829211> Checking if github.com/autobrr/go-rtorrent 1.0.1 needs updating
>   proxy | 2023/08/03 22:06:36 [592] GET https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/list
>   proxy | 2023/08/03 22:06:37 [592] 200 https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/list
>   proxy | 2023/08/03 22:06:37 [594] GET https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/v1.9.0.info
>   proxy | 2023/08/03 22:06:37 [594] 200 https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/v1.9.0.info
>   proxy | 2023/08/03 22:06:37 [596] GET https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/v1.9.0.mod
>   proxy | 2023/08/03 22:06:37 [596] 200 https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/v1.9.0.mod
>   proxy | 2023/08/03 22:06:37 [598] GET https://proxy.golang.org:443/sumdb/sum.golang.org/supported
>   proxy | 2023/08/03 22:06:37 [598] 404 https://proxy.golang.org:443/sumdb/sum.golang.org/supported
>   proxy | 2023/08/03 22:06:37 [600] GET https://sum.golang.org:443/lookup/github.com/autobrr/go-rtorrent@v1.9.0
>   proxy | 2023/08/03 22:06:37 [600] 200 https://sum.golang.org:443/lookup/github.com/autobrr/go-rtorrent@v1.9.0
>   proxy | 2023/08/03 22:06:37 [602] GET https://sum.golang.org:443/tile/8/0/x067/914
>   proxy | 2023/08/03 22:06:37 [602] 200 https://sum.golang.org:443/tile/8/0/x067/914
> updater | 2023/08/03 22:06:37 INFO <job_704829211> Latest version is 1.9.0
> ```
> 
> So it appears at the time the latest version was 1.9.0, even though I can see 1.10.0 there now: https://proxy.golang.org:443/github.com/autobrr/go-rtorrent/@v/list

Yeah, we ended up with a null tag to get around the bug. The expected behaviour was grouped updates would run our tests, and if something failed, backout individual updates until it passed. Most projects can't sustain 50 PRs to merge single deps. It's another issue I opened, but npm not even creating a PR is just completely bizarre behaviour.
